client.py

```diff
@@ -101,21 +101,28 @@ def filter_git_commit_data(repository_name, repository_team, repository_version)
     (e.g raise self.__createException(status, responseHeaders, output)
             github.GithubException.GithubException: 502 {'message': 'Server Error'}
     """
-    repo_dict = {}
-    number = 0
-    repository_path = repository_team + repository_name
-    repo_dict.update({number: {"lastChecked": str(datetime.utcnow()),
+    git_json_filename = "{}.json".format(repository_name)
+    try:
+        with open(current_dir + "/git_files/" + git_json_filename, "r") as commit_json:
+            repo_dict = json.load(commit_json)
+            number = len(repo_dict)
+            repo_dict.update({'0': {"lastChecked": str(datetime.utcnow()),
                                "last_two_releases": repository_version}})
+    except FileNotFoundError:
+        repo_dict = {}
+        repo_dict.update({'0': {"lastChecked": str(datetime.utcnow()),
+                               "last_two_releases": repository_version}})
+        number = 1
 
+    repository_path = repository_team + repository_name
     try:
         latest_release = datetime.strptime(repository_version['LatestRelease']['date'], '%a, %d %b %Y %H:%M:%S GMT')
         previous_release = datetime.strptime(repository_version['PreviousRelease']['date'], '%a, %d %b %Y %H:%M:%S GMT')
     except TypeError:
         previous_release = lastWeek
         latest_release = datetime.utcnow()
 
-    number += 1
-    for commit in git.get_repo(repository_path).get_commits(since=previous_release):
+    for commit in git.get_repo(repository_path).get_commits(since=lastWeek):
```

Comment:
This line does not look good for me because:
e.g. the commits from the last version have been in the last 3 weeks, so in this case only the commits since lastWeek will be saved
we need to use the previous release from now

Reply:
In this specific case, I used since purely to test the implementation. As far as I can tell , the logic in  this PR works. I tested it by modifying since variable a lot of time to detect bugs and see the behavior. "since=lastWeek" doesn't have place in the final version and after running the script for the first time bringing all the commits and store them in JSON format , my best guess is that "since=previous_release" will become "since=lastRun" .  

For testing purposes I had to run script multiple times in chronological order.